src/g1.rs

```diff
@@ -282,101 +330,12 @@ impl G1Affine {
-
-        res
-    }
-
-    /// Attempts to deserialize an uncompressed element. See [`notes::serialization`](crate::notes::serialization)
-    /// for details about how group elements are serialized.
-    pub fn from_uncompressed(bytes: &[u8; 96]) -> CtOption<Self> {
-        Self::from_uncompressed_unchecked(bytes)
-            .and_then(|p| CtOption::new(p, p.is_on_curve() & p.is_torsion_free()))
-    }
-
-    /// Attempts to deserialize an uncompressed element, not checking if the
-    /// element is on the curve and not checking if it is in the correct subgroup.
-    /// **This is dangerous to call unless you trust the bytes you are reading; otherwise,
-    /// API invariants may be broken.** Please consider using `from_uncompressed()` instead.
-    pub fn from_uncompressed_unchecked(bytes: &[u8; 96]) -> CtOption<Self> {
-        // Obtain the three flags from the start of the byte sequence
-        let compression_flag_set = Choice::from((bytes[0] >> 7) & 1);
-        let infinity_flag_set = Choice::from((bytes[0] >> 6) & 1);
-        let sort_flag_set = Choice::from((bytes[0] >> 5) & 1);
-
-        // Attempt to obtain the x-coordinate
-        let x = {
-            let mut tmp = [0; 48];
-            tmp.copy_from_slice(&bytes[0..48]);
-
-            // Mask away the flag bits
-            tmp[0] &= 0b0001_1111;
-
-            Fp::from_bytes(&tmp)
-        };
-
-        // Attempt to obtain the y-coordinate
-        let y = {
-            let mut tmp = [0; 48];
-            tmp.copy_from_slice(&bytes[48..96]);
-
-            Fp::from_bytes(&tmp)
-        };
-
-        x.and_then(|x| {
-            y.and_then(|y| {
-                // Create a point representing this value
-                let p = G1Affine::conditional_select(
-                    &G1Affine {
-                        x,
-                        y,
-                        infinity: infinity_flag_set,
-                    },
-                    &G1Affine::identity(),
-                    infinity_flag_set,
-                );
-
-                CtOption::new(
-                    p,
-                    // If the infinity flag is set, the x and y coordinates should have been zero.
-                    ((!infinity_flag_set) | (infinity_flag_set & x.is_zero() & y.is_zero())) &
-                    // The compression flag should not have been set, as this is an uncompressed element
-                    (!compression_flag_set) &
-                    // The sort flag should not have been set, as this is an uncompressed element
-                    (!sort_flag_set),
-                )
-            })
-        })
-    }
-
     /// Attempts to deserialize a compressed element. See [`notes::serialization`](crate::notes::serialization)
     /// for details about how group elements are serialized.
-    pub fn from_compressed(bytes: &[u8; 48]) -> CtOption<Self> {
+    pub fn from_bytes(bytes: &[u8; 48]) -> CtOption<Self> {
```

Comment:
Do we want this to return a `CtOption` still? I thought we were going to return the correct type directly.

Reply:
I don't want, but I believe this `CtOption` problem should be handled by a different issue